src/main/java/org/elasticsearch/plugins/PluginManager.java

```diff
@@ -19,674 +19,557 @@
 
-        Path distroFile(Environment env) {
-            return env.pluginsFile().resolve(name + ".zip");
-        }
+                } catch (Exception e) {
+                    terminal.printError("%s", e.getMessage());
+                    return ExitStatus.CANT_CREATE;
 
-        Path extractedDir(Environment env) {
-            return env.pluginsFile().resolve(name);
-        }
+                } finally {
+                    // Deletes the artifact
+                    Files.deleteIfExists(install.artifact());
+                }
+            }
 
-        Path binDir(Environment env) {
-            return env.homeFile().resolve("bin").resolve(name);
+            terminal.println("Plugin installed successfully!");
+            return ExitStatus.OK;
         }
+    }
 
-        Path configDir(Environment env) {
-            return env.configFile().resolve(name);
+    private static void validatePluginName(String pluginName) {
+        if (!Strings.hasText(pluginName)) {
+            throw new ElasticsearchIllegalArgumentException("Plugin's name cannot be null or empty");
+        }
+        if (!pluginName.matches("[a-zA-Z0-9_\\-]+")) {
+            throw new ElasticsearchIllegalArgumentException("Plugin's name is invalid");
         }
 
-        static PluginHandle parse(String name) {
-            String[] elements = name.split("/");
-            // We first consider the simplest form: pluginname
-            String repo = elements[0];
-            String user = null;
-            String version = null;
-
-            // We consider the form: username/pluginname
-            if (elements.length > 1) {
-                user = elements[0];
-                repo = elements[1];
-
-                // We consider the form: username/pluginname/version
-                if (elements.length > 2) {
-                    version = elements[2];
-                }
-            }
-
-            if (repo.startsWith("elasticsearch-")) {
-                // remove elasticsearch- prefix
-                String endname = repo.substring("elasticsearch-".length());
-                return new PluginHandle(endname, version, user, repo);
+        for (String name : LocalPluginRepository.FORBIDDEN_NAMES) {
+            if (name.equalsIgnoreCase(pluginName)) {
+                throw new ElasticsearchIllegalArgumentException("Plugin's name cannot be equal to '" + pluginName + "'");
             }
+        }
+    }
 
-            if (name.startsWith("es-")) {
-                // remove es- prefix
-                String endname = repo.substring("es-".length());
-                return new PluginHandle(endname, version, user, repo);
-            }
+    /**
+     * Prints "no version" when the version is null or empty.
+     *
+     * @param version the plugin's version
+     * @return plugin's version or "no version"
+     */
+    private static String checkVersion(String version) {
+        return Strings.hasText(version) ? version : "no version";
```

Comment:
In other places of the code, we used `NA` instead. For example, `_cat/plugins` API returns `NA`. I think we should keep that consistent.